### PR TITLE
[TGUI] Airlock Controller -> TGUI, Remove Advanced Airlock Controller

### PR DESCRIPTION
**Pull request by variableundefined**
## What Does This PR Do
Updated Airlock controllers to TGUI. There's some minor cosmetics change:
- There's now Actions and Information sections in the new UI
- The progress bar fill up at 1013 KPA (Which is pretty atypical for airlock) instead of whatever the old number was. If requested I'll change back to 303 or somewhere closer. 
- There's some minor spacing changes

In addition, I removed the code and template for Advanced_Airlock_Console. Advanced airlock controller / console was not used anywhere by any airlock, and I was tripped up by the original TGUI conversion list requesting me to changed advanced_airlock_console.tmpl, only to realize it was in fact, an unused template and only the simple console is used. 

I don't see any need to maintain two version of the same template or code with one being completely unused.

## Images of changes
**Old:**
image
**New:**
image

## Changelog
:cl:
add: Airlock controllers has been updated to TGUI.
del: Unused Advanced Airlock Controller were removed.
/:cl:

### CODE REVIEW

**Comment by TDSSS** [comment]
Old and new seem identical here (down to cursor position). Is it the right picture?

File: code/game/machinery/embedded_controller/embedded_controller_base.dm [hunk 1]
```diff
@@ -27,18 +27,18 @@
...
 
 /obj/machinery/embedded_controller/attack_hand(mob/user as mob)
 	if(!user.IsAdvancedToolUser())
 		return 0
-	src.ui_interact(user)
+	tgui_interact(user)
 
-/obj/machinery/embedded_controller/ui_interact()
-	return
+/obj/machinery/embedded_controller/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
```
Comment (AffectedArc07) [inline code/game/machinery/embedded_controller/embedded_controller_base.dm:40]: If this proc just calls parent, just remove it

File: tgui/yarn.lock [hunk 2]
```diff
@@ -1500,9 +1500,9 @@ caniuse-api@^3.0.0:
     lodash.uniq "^4.5.0"
 
 caniuse-lite@^1.0.0, caniuse-lite@^1.0.30001038:
-  version "1.0.30001042"
-  resolved "https://registry.yarnpkg.com/caniuse-lite/-/caniuse-lite-1.0.30001042.tgz#c91ec21ec2d270bd76dbc2ce261260c292b8c93c"
-  integrity sha512-igMQ4dlqnf4tWv0xjaaE02op9AJ2oQzXKjWf4EuAHFN694Uo9/EfPVIPJcmn2WkU9RqozCxx5e2KPcVClHDbDw==
+  version "1.0.30001148"
```
Comment (AffectedArc07) [inline tgui/yarn.lock:1503]: Please revert the changes to this

File: code/game/machinery/embedded_controller/airlock_controllers.dm [hunk 3]
```diff
@@ -91,49 +36,47 @@
 		tag_chamber_sensor = given_tag_chamber_sensor
 	..()
 
-/obj/machinery/embedded_controller/radio/airlock/airlock_controller/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/embedded_controller/radio/airlock/airlock_controller/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	user.set_machine(src)
```
Comment (AffectedArc07) [inline code/game/machinery/embedded_controller/airlock_controllers.dm:40]:
```suggestion

```

File: code/game/machinery/embedded_controller/airlock_controllers.dm [hunk 4]
```diff
@@ -91,49 +36,47 @@
...
 		return
 
-	usr.set_machine(src)
-	src.add_fingerprint(usr)
+	add_fingerprint(usr)
 
-	var/clean = 0
-	switch(href_list["command"])	//anti-HTML-hacking checks
+	var/clean = FALSE
+	switch(action)	//anti-HTML-hacking checks
```
Comment (AffectedArc07) [inline code/game/machinery/embedded_controller/airlock_controllers.dm:64]: I feel like this can be replaced with 
```
if(action in list("cycle_ext", "cycle_int", etc)

**Comment by variableundefined** [comment]
@TDSSS updated image, uploaded wrong one by accident

File: code/game/machinery/embedded_controller/airlock_controllers.dm [hunk 5]
```diff
@@ -91,49 +36,34 @@
 		tag_chamber_sensor = given_tag_chamber_sensor
 	..()
 
-/obj/machinery/embedded_controller/radio/airlock/airlock_controller/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/embedded_controller/radio/airlock/airlock_controller/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		ui = new(user, src, ui_key, "simple_airlock_console.tmpl", name, 470, 290)
+		ui = new(user, src, ui_key, "Airlock", name, 470, 290, master_ui, state)
```
Comment (SteelSlayer) [inline code/game/machinery/embedded_controller/airlock_controllers.dm:42]: I would suggest a more descriptive name for this, and I also recommend `AirlockController.js` as the file name too.
```suggestion
		ui = new(user, src, ui_key, "AirlockController", name, 470, 290, master_ui, state)
```

Comment (AffectedArc07) [reply code/game/machinery/embedded_controller/airlock_controllers.dm:42]: Siding with steel. Please change this, possibly to `ExternalAirlockController` since it managed atmos stuffs.

File: code/game/machinery/embedded_controller/embedded_controller_base.dm [hunk 6]
```diff
@@ -27,18 +27,15 @@
...
-	src.ui_interact(user)
+	tgui_interact(user)
 
 /obj/machinery/embedded_controller/attack_ai(mob/user as mob)
-	src.ui_interact(user)
+	tgui_interact(user)
 
 /obj/machinery/embedded_controller/attack_hand(mob/user as mob)
 	if(!user.IsAdvancedToolUser())
 		return 0
```
Comment (AffectedArc07) [inline code/game/machinery/embedded_controller/embedded_controller_base.dm:37]:
```suggestion
		return FALSE
```

File: tgui/packages/tgui/interfaces/Airlock.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+      <Window.Content>
+        <Section title="Information">
+          <LabeledList>
+            <LabeledList.Item label="Chamber Pressure">
+              <ProgressBar
+                color={getStatusColor(data.chamber_pressure)}
+                value={data.chamber_pressure}
+                minValue={MIN_BAR_VALUE}
+                maxValue={MAX_BAR_VALUE}>
+                {data.chamber_pressure} kPa
```
Comment (AffectedArc07) [inline tgui/packages/tgui/interfaces/Airlock.js:51]:
```suggestion
                color={getStatusColor(chamber_pressure)}
                value={chamber_pressure}
                minValue={MIN_BAR_VALUE}
                maxValue={MAX_BAR_VALUE}>
                {chamber_pressure} kPa
```
If youre defining these off of data at line 34-39, you dont need to include `data.` in these statements

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/ExternalAirlockController.js [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+      <Window.Content>
+        <Section title="Information">
+          <LabeledList>
+            <LabeledList.Item label="Chamber Pressure">
+              <ProgressBar
+                color={getStatusColor(chamber_pressure)}
+                value={chamber_pressure}
+                minValue={MIN_BAR_VALUE}
+                maxValue={MAX_BAR_VALUE}>
+                {data.chamber_pressure} kPa
```
Comment (AffectedArc07) [inline tgui/packages/tgui/interfaces/ExternalAirlockController.js:51]:
```suggestion
                {chamber_pressure} kPa
```

File: tgui/packages/tgui/interfaces/ExternalAirlockController.js [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+              content={"Cycle to Interior"}
+              icon={"arrow-circle-right"}
+              disabled={processing}
+              onClick={() => act("cycle_int")} />
+          </Box>
+          <Box>
+            <Button
+              content={"Force Exterior Door"}
+              icon={"exclamation-triangle"}
+              color={interior_status === "open" ? "red" : data.processing ? "yellow": null}
```
Comment (AffectedArc07) [inline tgui/packages/tgui/interfaces/ExternalAirlockController.js:73]:
```suggestion
              color={interior_status === "open" ? "red" : processing ? "yellow": null}
```

File: tgui/packages/tgui/interfaces/ExternalAirlockController.js [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+              onClick={() => act("force_ext")} />
+            <Button
+              content={"Force Interior Door"}
+              icon={"exclamation-triangle"}
+              color={interior_status === "open" ? "red" : data.processing ? "yellow": null}
```
Comment (AffectedArc07) [inline tgui/packages/tgui/interfaces/ExternalAirlockController.js:78]:
```suggestion
              color={interior_status === "open" ? "red" : processing ? "yellow": null}
```

File: tgui/packages/tgui/interfaces/ExternalAirlockController.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+              onClick={() => act("force_int")} />
+          </Box>
+          <Box>
+            <Button
+              content={"Abort"}
+              icon={"ban"}
+              color={"red"}
+              disabled={!data.processing}
```
Comment (AffectedArc07) [inline tgui/packages/tgui/interfaces/ExternalAirlockController.js:86]:
```suggestion
              disabled={!processing}
```

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
Last things

**Review by AffectedArc07 — APPROVED** [review]
